queue/driver_mongo.go

```diff
@@ -1333,54 +1231,24 @@ func (d *mongoDriver) Next(ctx context.Context) amboy.Job {
 			return nil
 		case <-timer.C:
 			misses++
-
-			iter, err := d.getNextCursor(ctx)
+			job, dispatchInfo, err := d.dispatchStaleAndPending(ctx, startAt)
 			if err != nil {
-				grip.Warning(message.WrapError(err, message.Fields{
-					"message":       "problem finding jobs ready to dispatch",
-					"driver_id":     d.instanceID,
-					"service":       "amboy.queue.mdb",
-					"operation":     "retrieving next job",
-					"is_group":      d.opts.UseGroups,
-					"group":         d.opts.GroupName,
-					"duration_secs": time.Since(startAt).Seconds(),
-				}))
-				return nil
-			}
-
-			job, dispatchInfo := d.tryDispatchJob(ctx, iter, startAt)
-			dispatchMisses += dispatchInfo.misses
-			dispatchSkips += dispatchInfo.skips
-			if err = iter.Err(); err != nil {
 				if job != nil {
```

Comment:
Seems like this could never happen since the function returns either `err` or `job`, but not both.

Reply:
`dispatchNextJob` returns the job now